File: robin/java/src/programmers/level1/study6/RemainZero.java
package programmers.level1.study6;

import java.util.Arrays;

public class RemainZero {
    public static int[] solution(int[] arr, int divisor) {
        int [] answer = Arrays.stream(Arrays.stream(arr).filter(x -> x % divisor == 0).toArray()).sorted().toArray();

        if (answer.length == 0) return new int[] {-1};

        return answer;
    }

    public static void main(String[] args) {
        int[] arr = {5, 9, 7, 10};
        int divisor = 5;
        System.out.println(Arrays.toString(solution(arr, divisor)));
    }
}

File: robin/python/programmers/level1/study6/remain_zero.py
def solution(arr, divisor):
    answer = []
    for i in arr:
        if i % divisor == 0:
            answer.append(i)

    if len(answer) == 0:
        answer.append(-1)

    return sorted(answer)


def solution1(arr, divisor):
    return sorted([n for n in arr if n % divisor == 0]) or [-1]


if __name__ == '__main__':
    arr = [2, 36, 1, 3]
    divisor = 1
    print(solution(arr, divisor))
    print(solution1(arr, divisor))
